extras/iris_pipeline_project/pipeline/ovmsmodel.py

```diff
@@ -0,0 +1,132 @@
+import os
+import json
+import numpy as np
+import torch
+import torch.nn as nn
+import joblib
+from pyovms import Tensor
+from model import LogisticRegressionTorch, KMeansSkLearn, ModelClass
+from sklearn.preprocessing import LabelEncoder
+from sklearn.metrics import accuracy_score, precision_score, recall_score, f1_score
+
+
+MODEL_PATH = "/workspace/model/generic_model/model.pt"
+ENCODER_PATH = "/workspace/model/generic_model/label_encoder.joblib"
+META_PATH = "/workspace/model/generic_model/meta.json"
+
+AVAILABLE_MODEL_CLASSES = {
+    "LogisticRegressionTorch": LogisticRegressionTorch,
+    "KMeansSkLearn": KMeansSkLearn
+}
+
+class OvmsPythonModel:
+    def initialize(self, kwargs):
+        print("[initialize] Python node initialized", flush=True)
+        self.model_obj = None
+
+    def execute(self, inputs):
+        try:
+            inp_bytes = bytes(inputs[0].data)
+            first_brace = inp_bytes.find(b'{')
+            if first_brace > 0:
+                inp_bytes = inp_bytes[first_brace:]
+            payload = json.loads(inp_bytes.decode("utf-8"))
+
+            mode = payload.get("mode")
+            X = np.array(payload.get("X"), dtype=np.float32)
+            y = payload.get("y", None)
+            params = payload.get("params", {})
+            model_class_name = payload.get("model_class", "LogisticRegressionTorch")
+
+            if model_class_name not in AVAILABLE_MODEL_CLASSES:
+                raise ValueError(f"Unknown model: {model_class_name}")
+
+            model_obj = AVAILABLE_MODEL_CLASSES[model_class_name]()
+
+            if model_class_name == "KMeansSkLearn":
```

Comment:
I don't like this conditional logic. `execute` method should be generic and wrap all possible implementations. Having any model specific code goes against this idea. Please move those implementation specific code blocks to its classes. 
You define the interface and convention. You can make `fit` or `infer` method return not only results, but also metrics that can be used directly to create pyovms.Tensor object like:
```
trained_model, training_metrics = model_obj.fit(X, y, params))
...
return [Tensor("pipeline_output", training_metrics)]
```